update dependencies

mimmi20 opened the pull request.

mimmi20 committed:
update dependencies

dependabot[bot] committed:
github-actions: bump actions/upload-artifact from 3 to 4

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 3 to 4.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: '4'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
github-actions: bump actions/cache from 3 to 4

Bumps [actions/cache](https://github.com/actions/cache) from 3 to 4.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-version: '4'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
github-actions: bump actions/checkout from 3 to 4

Bumps [actions/checkout](https://github.com/actions/checkout) from 3 to 4.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '4'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mimmi20 committed:
add some workflows

mimmi20 committed:
add some config files

mimmi20 committed:
Update readme.md

mimmi20 committed:
Merge pull request #4 from mimmi20/dependabot/github_actions/master/actions/checkout-4

github-actions: bump actions/checkout from 3 to 4

dependabot[bot] committed:
github-actions: bump actions/setup-node from 3 to 4

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 3 to 4.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-version: '4'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mimmi20 committed:
Merge pull request #3 from mimmi20/dependabot/github_actions/master/actions/cache-4

github-actions: bump actions/cache from 3 to 4

mimmi20 committed:
Merge pull request #2 from mimmi20/dependabot/github_actions/master/actions/setup-node-4

github-actions: bump actions/setup-node from 3 to 4

mimmi20 committed:
Merge pull request #1 from mimmi20/dependabot/github_actions/master/actions/upload-artifact-4

github-actions: bump actions/upload-artifact from 3 to 4

mimmi20 committed:
update readme

mimmi20 committed:
remove duplicate blank lines

mimmi20 committed:
change import

mimmi20 committed:
add some configs

mimmi20 committed:
skip some tasks

mimmi20 committed:
add typescript

dependabot[bot] committed:
npm (base)(deps-dev): bump ava from 6.3.0 to 6.4.0

Bumps [ava](https://github.com/avajs/ava) from 6.3.0 to 6.4.0.
- [Release notes](https://github.com/avajs/ava/releases)
- [Commits](https://github.com/avajs/ava/compare/v6.3.0...v6.4.0)

---
updated-dependencies:
- dependency-name: ava
  dependency-version: 6.4.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
npm (base)(deps-dev): bump xo from 1.0.5 to 1.1.0

Bumps [xo](https://github.com/xojs/xo) from 1.0.5 to 1.1.0.
- [Release notes](https://github.com/xojs/xo/releases)
- [Commits](https://github.com/xojs/xo/compare/v1.0.5...v1.1.0)

---
updated-dependencies:
- dependency-name: xo
  dependency-version: 1.1.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mimmi20 committed:
Merge pull request #6 from mimmi20/dependabot/npm_and_yarn/master/ava-6.4.0

npm (base)(deps-dev): bump ava from 6.3.0 to 6.4.0

mimmi20 committed:
Merge pull request #7 from mimmi20/dependabot/npm_and_yarn/master/xo-1.1.0

npm (base)(deps-dev): bump xo from 1.0.5 to 1.1.0

mimmi20 committed:
remove nodejs 23

mimmi20 committed:
Merge pull request #5 from mimmi20/updates

add some workflows

mimmi20 committed:
update install script

mimmi20 committed:
update install script

mimmi20 committed:
update binaries

mimmi20 committed:
Merge pull request #8 from mimmi20/updates

update install script

mimmi20 committed:
Update dependabot.yml

mimmi20 committed:
Merge pull request #9 from mimmi20/updates

Update dependabot.yml

vHeemstra commented on package.json:
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@vheemstra/avifenc-bin",
   "version": "2.1.1",
```
Version should be 3.0.0, since we're upgrading to libavif 1.3.0, which is a major version change.

vHeemstra replied:
This might be unnecessary, since the GitHub Action should trigger `npm publish` on new releases. But I might be mistaken. (It's been a while since I worked on the repo..)

vHeemstra commented on .github/workflows/manual_build.yml:
```diff
@@ -60,7 +60,7 @@ jobs:
       # - name: macOS - Restore Homebrew deps from cache
```
We probably should uncomment these steps to use Homebrew deps caching and to solve the action lint errors regarding missing `try-homebrew-cache` and `cache-homebrew-deps-restore` variables.

vHeemstra commented on .github/workflows/manual_build.yml:
```diff
@@ -105,21 +105,19 @@ jobs:
       # - name: macOS - Save Homebrew deps to cache
```
This should be uncommented as well.

vHeemstra reviewed (changes requested):
Great work, thank you! I put a couple of comments where changes are needed. I would do it myself, but I couldn't change those. (Since I couldn't push changes to your PR master branch.)

dependabot[bot] committed:
npm (base)(deps-dev): bump xo in the development-dependencies group

Bumps the development-dependencies group with 1 update: [xo](https://github.com/xojs/xo).


Updates `xo` from 1.1.0 to 1.1.1
- [Release notes](https://github.com/xojs/xo/releases)
- [Commits](https://github.com/xojs/xo/compare/v1.1.0...v1.1.1)

---
updated-dependencies:
- dependency-name: xo
  dependency-version: 1.1.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #10 from mimmi20/dependabot/npm_and_yarn/master/development-dependencies-99781923f6

npm (base)(deps-dev): bump xo from 1.1.0 to 1.1.1 in the development-dependencies group

dependabot[bot] committed:
github-actions: bump Dunning-Kruger/lock-issues

Bumps the production-dependencies group with 1 update: [Dunning-Kruger/lock-issues](https://github.com/dunning-kruger/lock-issues).


Updates `Dunning-Kruger/lock-issues` from 1.1 to 1.2
- [Release notes](https://github.com/dunning-kruger/lock-issues/releases)
- [Commits](https://github.com/dunning-kruger/lock-issues/compare/v1.1...v1.2)

---
updated-dependencies:
- dependency-name: Dunning-Kruger/lock-issues
  dependency-version: '1.2'
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: production-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #11 from mimmi20/dependabot/github_actions/master/production-dependencies-c999bf36e3

github-actions: bump Dunning-Kruger/lock-issues from 1.1 to 1.2 in the production-dependencies group

mimmi20 commented:
> Great work, thank you! I put a couple of comments where changes are needed. I would do it myself, but I couldn't change those. (Since I couldn't push changes to your PR master branch.)

The Protection is removed now

mimmi20 committed:
remove some files

mimmi20 committed:
update workflow

mimmi20 committed:
update dependabot config

mimmi20 committed:
update workflow

mimmi20 committed:
update dependencies

mimmi20 commented:
@vHeemstra Please look over my changes. A lot of them are a personell thing, like the added workflows, replacing tabs with spaces, ...
And I not get the compiling to work.

vHeemstra commented:
After some deliberation, I decided to overhaul the entire package myself. I haven't been updating this package in quite a while, so it needed that!

Choosing less is more as my approach, I stripped the project of all old, unmaintained and obsolete dependencies. I replaced this all with regular Node.JS functionality.

Have a look at the new release v3.0.0 🥳

Since this project is not that big and does not update that often, I chose to hold back on all GitHub and other workflow related config stuff. It seems overkill in this case.

Anyway, thanks for prompting me to have a look at this package and if you see any missing functionality, please feel free to open a new PR.

Have a look at the new release v3.0.0
